### PR TITLE
ENH: Added mongo shell authentication option

### PR DESCRIPTION
**Pull request by arkilic**
Use --auth with --mongo-user <user> --mongo-pwd <pwd> options for authentication enforced mongo shells and --no-auth for open shells.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+PORT=$(manage-iocs nextport)
+HOST=$HOSTNAME
+USER=tornado" > config
+
+echo '#!/bin/bash
+export PATH=/opt/conda/bin:$PATH
+source activate services
+python /services/amostra/startup.py --mongo-host=localhost --mongo-port=27017 --database=amostra --service-port=7770 --timezone=US/Eastern' > launch.sh
+
+
```
Comment (hhslepicka) [inline deploy.sh]: Missing here the:
```
chmod +x launch.sh
```
The service will fail to go up without it.

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+apt-get install sysv-rc-softioc
+if [ ! -d /opt/conda_envs/services/ ]; then conda create -p /opt/conda_envs/services python=3.5; fi
+source activate services
+conda install pymongo ujson tornado jsonschema yaml pytz doct
+pip install requests
+mkdir /services
+cd /services
+mkdir -p /epics/iocs/
+
+t clone https://github.com/NSLS-II/amostra.git
```
Comment (hhslepicka) [inline deploy.sh]: Typo
```
git clone
```

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+echo "NAME=amostra
+PORT=$(manage-iocs nextport)
+HOST=$HOSTNAME
+USER=tornado" > config
+
+echo '#!/bin/bash
+export PATH=/opt/conda/bin:$PATH
+source activate services
+python /services/amostra/startup.py --mongo-host=localhost --mongo-port=27017 --database=amostra --service-port=7770 --timezone=US/Eastern' > launch.sh
```
Comment (hhslepicka) [inline deploy.sh]: Missing the new parameters for auth. The question is: How to deal with it in this case here? =(

**Comment by hhslepicka** [comment]
:+1: Looks pretty good. I just added a couple of comments at the deploy.sh.

**Comment by arkilic** [comment]
I should separate the deployment script from this PR. I am curious if we can use it to deploy servers in travis. Gotta find out

**Comment by hhslepicka** [comment]
You mean doing it on a shell script and not inside of .travis.yml?

**Comment by arkilic** [comment]
Using deploy.sh inside travis like init-mds or init-fs  in nsls2-ci repo.

**Comment by hhslepicka** [comment]
In this case you also need to ensure that the NSLS2 Debian mirror is configured at the machine before the installation of the **_sysv-rc-softioc_** package.

Here is the code but maybe you will need to do some tweaking for the distro name:
```
    - wget --quiet http://epics.nsls2.bnl.gov/debian/repo-key.pub -O - | sudo apt-key add -
    - echo "deb http://epics.nsls2.bnl.gov/debian/ jessie/staging main contrib" | sudo tee /etc/apt/sources.list.d/nsls2.list
    - sudo apt-get update
```

**Comment by arkilic** [comment]
Yup! courtesy of @mdavidsaver

**Comment by hhslepicka** [comment]
This will probably cover it:
```
echo "This: `lsb_release -c -s`"
```
I updated the code in the comment.

**Pull request merged**